storage/writer.go

```diff
@@ -637,3 +642,11 @@ func read(r io.Reader, buf []byte) (int, bool, error) {
 	}
 	return recvd, done, err
 }
+
+func checkCanceled(err error) error {
+	if status.Code(err) == codes.Canceled {
```

Comment:
This seems like the right thing to do, but just to check: are we ever losing any information here by obscuring the difference between a user-side context cancelation and the RPC call returning `codes.Canceled`?

Reply:
We _might_ be, but I'm not entirely sure. I don't think we would be since this error is induced by client-side request settings i.e. the user sets a deadline on the context, and the server notices that the deadline it received has expired. Nothing for the server to add there imo.

One other option would be to change how `gax.APIError` handles `codes.Canceled` and instead of storing the original grpc-go error, it stores a `context.Canceled` after parsing the `Status`. But that's a bit more far reaching and I'd be hesitant to do that.